src/plugins/intel_gpu/src/graph/include/kernel_selector_helper.h

```diff
@@ -106,6 +108,54 @@ kernel_selector::weights_tensor convert_weights_tensor(const layout& l, bool is_
 layout from_weights_tensor(const kernel_selector::weights_tensor& t);
 kernel_selector::activation_function get_kernel_selector_activation_param(activation_func activation_func);
 
+struct kernel_impl_params {
+    bool has_runtime_layouts = false;
+    const program& prog;
+    std::shared_ptr<const primitive> desc;
+    size_t unique_id;
+    std::vector<layout> input_layouts;
+    layout output_layout;
+    std::vector<cldnn::fused_primitive_desc> fused_desc;
+    std::vector<activation_func> fused_act_funcs;
+    std::vector<activation_additional_params> activation_params;
+
+    layout weights_layout = layout(data_types::f32, format::any, tensor());
+
+    bool bias_term = false;
+    layout bias_layout = layout(data_types::f32, format::any, tensor());
+
+    bool weights_zero_points_term = false;
+    layout weights_zero_points_layout = layout(data_types::f32, format::any, tensor());
+
+    bool activations_zero_points_term = false;
+    layout activations_zero_points_layout = layout(data_types::f32, format::any, tensor());
+
+    bool compensation_term = false;
+    layout compensation_layout = layout(data_types::f32, format::any, tensor());
+
+    kernel_impl_params(program& _prog, std::shared_ptr<const primitive> _desc, size_t _uid,
+                       const std::vector<layout>& _int_layouts, layout _out_layout,
+                       const std::vector<cldnn::fused_primitive_desc>& _fused_descs,
+                       const std::vector<activation_func>& _fused_act_funcs, const std::vector<activation_additional_params>& _act_params,
+                       layout _weights_layout = layout(data_types::f32, format::any, tensor()),
+                       bool   _bias_term = false,                    layout _bias_layout = layout(data_types::f32, format::any, tensor()),
```

Comment:
We have an `optional_data_type` wrapper for similar usage scenarios, so maybe it makes sense to introduce `optional_layout` which will combine presence flag with the data itself. 
Ideally, we need a template class similar to std::optional with some helpers on top of that. And if we transition from c++11 to c++17 at some point, we'll just replace custom implementation with the standard one.

Reply:
Good point, I applied it!